waspc/e2e-test/GoldenTest.hs

```diff
@@ -83,3 +99,24 @@ runGoldenTest goldenTest = do
     writeFileManifest baseAbsFp filePaths manifestAbsFp = do
       let sortedRelativeFilePaths = unlines . sort . map (makeRelative baseAbsFp) $ filePaths
       writeFile manifestAbsFp sortedRelativeFilePaths
+
+    -- This function normalizes all package.json files for comparison as `npm install` can overwrite
+    -- them when creating/updating package-lock.json. Different versions of npm may produce different
+    -- (but semantically equivalent) package.json files, thus triggering false positives.
+    -- Ref: https://github.com/wasp-lang/wasp/issues/482
+    reformatPackageJsonFiles :: [FilePath] -> IO ()
+    reformatPackageJsonFiles allOutputFilePaths = do
+      let packageJsonFilePathSuffix = FP.pathSeparator : "package.json"
+      let packageJsonFilePaths = filter (packageJsonFilePathSuffix `isSuffixOf`) allOutputFilePaths
+      mapM_ reformatJson packageJsonFilePaths
+      where
+        aesonPrettyConfig = Config {confIndent = Spaces 4, confCompare = compare, confNumFormat = Generic, confTrailingNewline = True}
+        reformatJson jsonFilePath = do
```

Comment:
I played a bit with golfing this one, I came up with
```hs
reformatJson jsonFilePath =
  writeFile jsonFilePath . encodePretty' aesonPrettyConfig . unsafeDecodeAnyJson =<< readFile jsonFilePath
  where unsafeDecodeAnyJson :: String -> Value
        unsafeDecodeAnyJson = fromJust . decode
```
assuming readFile is strict and not lazy, but I didn't compile it nor am I sure how much of an improvement is it -> this is more for fun.

Reply:
Check my comment above.

Reply:
Your golf was quite close (impressive!) :D 

```haskell
        reformatJson :: FilePath -> IO ()
        reformatJson jsonFilePath =
          BSL.writeFile jsonFilePath . AesonPretty.encodePretty' aesonPrettyConfig . unsafeDecodeAnyJson =<< B.readFile jsonFilePath
          where
            unsafeDecodeAnyJson :: B.ByteString -> Aeson.Value
            unsafeDecodeAnyJson = fromJust . Aeson.decodeStrict
```

I can read in strict with `Data.ByteString` but still need to write with `Data.ByteString.Lazy` as that is what `encodePretty'` returns. Even with the 1 in a million of dying during a write, I'll stick with this cleaned up version. Looks much nicer haha thanks

Reply:
That is ok! As long as you read with strict, you know you have read all the content at that point and you are good to go, there shouldn't be any issues. You won't be writing in the same file while reading from it, that is important.

Reply:
Yep, indeed! This does solve the problem of reading fully before starting to (over)write. My "1 in a million" hypothetical just meant _any_ disk write in general that could fail partway through. I wasn't sure if there was some OS/driver/other runtime error when it had started writing, but not yet fully done, that could ever leave a partially written file on error (I think this is possible?)... so not specific to this case but just a "what if a file write operation dies midway through" general question. The waiting until fully and successfully written, and only then atomically moving the file on top, does theoretically guard against that case. But it is a pretty pathological case I don't 100% know if/how it can occur, and if so, is really, really unlikely. The things I worry about sometimes lol ugh 😅

Reply:
Ah got it, that is a completely valid worry! But I agree that in this case it is probably not a big deal :). Still good to think about it!